api/http/proxy/manager.go

```diff
@@ -54,6 +56,34 @@ func (manager *Manager) CreateAndRegisterProxy(endpoint *portainer.Endpoint) (ht
 	return proxy, nil
 }
 
+// CreateAndRegisterProxy creates a new HTTP reverse proxy and adds it to the registered proxies.
+// It can also be used to create a new HTTP reverse proxy and replace an already registered proxy.
+func (manager *Manager) CreateAndRegisterRegistryProxy(registry *portainer.Registry) (http.Handler, error) {
+	var proxy http.Handler
+
+	registryURL, err := url.Parse("http://" + registry.URL)
```

Comment:
We should also be able to handle HTTPS registries here. I believe that we need to work on that.

Reply:
I thinks it's related to #1467 
Maybe the way to go will be to add a new parameter `protocol` to Registry object
When the validation occurs, this parameter is automatically setted
Then, in this code, we use this parameter for the protocol

Reply:
Hmm I'm thinking about it, maybe the protocol should be part of the URL ? I wonder if it would have an impact on the existing implementation (which is subject to change, see https://github.com/portainer/portainer/issues/1434#issuecomment-347270725

Reply:
FYI : To pull an image, protocol must be _not_ present

Reply:
Ok, we're going to support HTTP only for now. 

We'll tackle protocol support in #1467